api/src/opentrons/protocol_engine/state/addressable_areas.py

```diff
@@ -462,6 +470,30 @@ def get_addressable_area_center(self, addressable_area_name: str) -> Point:
             z=position.z,
         )
 
+    def get_fixture_by_deck_slot_name(
+        self, slot_name: DeckSlotName
+    ) -> Optional[CutoutFixture]:
+        """Get the Cutout Fixture of a fixture currently loaded into a specific Deck Slot."""
+        deck_config = self.state.deck_configuration
+        if deck_config:
+            slot_cutout_id = DECK_SLOT_TO_CUTOUT_MAP[slot_name]
+            slot_cutout_fixture = None
+            # This will only ever be one under current assumptions
+            for cutout_id, cutout_fixture_id in deck_config:
+                if cutout_id == slot_cutout_id:
+                    slot_cutout_fixture = (
+                        deck_configuration_provider.get_cutout_fixture(
+                            cutout_fixture_id, self.state.deck_definition
+                        )
+                    )
+                    break
+            if slot_cutout_fixture is None:
+                raise CutoutDoesNotExistError(
+                    f"No Cutout was found in the Deck that matched provided slot {slot_name}."
+                )
```

Comment:
Would this comment be correct?

```suggestion
                # If this happens, it's a bug. Either DECK_SLOT_TO_CUTOUT_MAP
                # is missing an entry for the slot, or the deck configuration is missing
                # an entry for the cutout.
                raise CutoutDoesNotExistError(
                    f"No Cutout was found in the Deck that matched provided slot {slot_name}."
                )
```

Reply:
Thats accurate. This raise case is here to protect against the currently impossible. There is no way with our current concept of deck configuration to have a deck slot without a cutout fixture, nor to provide a deckslot here that isn't included within the currently provided selection of known deckslots to cutouts. However, on the off chance that information like that was provided we will catch it here. Hypothetically, if we decide someday you can load a deck with only some of its fixtures and the rest left literally empty (as in a big hole in the deck), this would catch instances where entities calling this function attempt to reference the "holes" in the deck.